contracts/validators/CredentialAtomicQueryValidatorBase.sol

```diff
@@ -160,6 +170,15 @@ abstract contract CredentialAtomicQueryValidatorBase is
     }
 
     function _checkClaimNonRevState(uint256 _id, uint256 _claimNonRevState) internal view {
+        // for privado identity and genesis state we don't need to check for expiration
+        if (_isPrivadoId(_id)) {
+            require(
+                GenesisUtils.isGenesisState(_id, _claimNonRevState),
+                "Privado identity is not genesis"
+            );
+            return;
+        }
+
```

Comment:
I think we should check if the contract support the id type.
```
s.state.getIdTypeIfSupported(_id);
```

Reply:
it is checked on getStateInfoById or Exists call